apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ValueConversion.cs

```diff
@@ -92,139 +108,108 @@ public static System.Type GetDefaultClrTypeFromSpannerType(this TypeCode code)
             switch (code)
             {
                 case TypeCode.Bool:
-                    return typeof(bool);
+                    return typeof(bool?);
                 case TypeCode.Int64:
-                    return typeof(long);
+                    return typeof(long?);
                 case TypeCode.Float64:
-                    return typeof(double);
+                    return typeof(double?);
                 case TypeCode.Timestamp:
                 case TypeCode.Date:
-                    return typeof(DateTime);
+                    return typeof(DateTime?);
                 case TypeCode.String:
                     return typeof(string);
                 case TypeCode.Bytes:
                     return typeof(byte[]);
                 case TypeCode.Array:
-                    return typeof(IList);
+                    return typeof(ArrayList);
                 case TypeCode.Struct:
-                    return typeof(IDictionary);
+                    return typeof(Hashtable);
```

Comment:
Something like `Dictionary<string, object>`? (I'm not clear exactly what a `Struct` means)